assets/src/dashboard/app/views/savedTemplates/content/index.js

```diff
@@ -58,27 +58,25 @@ function Content({
     'enableInProgressStoryActions'
   );
 
-  const handleMenuItemSelected = useCallback(
-    (sender, template) => {
-      setContextMenuId(-1);
-
-      switch (sender.value) {
-        case SAVED_TEMPLATE_CONTEXT_MENU_ACTIONS.OPEN_IN_EDITOR:
-          trackEvent('use_saved_template', {
-            name: template.title,
-            template_id: template.id,
-          });
-          actions.createStoryFromTemplate(template);
-          break;
-
-        case SAVED_TEMPLATE_CONTEXT_MENU_ACTIONS.PREVIEW:
-          actions.previewTemplate(null, template);
-          break;
+  // TODO add context menu actions
+  const handleCreateStoryFromTemplate = useCallback(
+    (template) => {
+      trackEvent('use_saved_template', {
+        name: template.title,
+        template_id: template.id,
+      });
+      actions.createStoryFromTemplate(template);
+    },
+    [actions]
+  );
```

Comment:
I'm not sure it's necessary to split these up.

The reason I split these up is because each `item` in the context menu takes its own `onClick` event handler now (rather than one big one that handles each item individually).

I split them up to follow a good separation of concerns, but we still need to divvy them up to each menu item. That's still split up by the `value` of the item. Ex:
```
menuItemActions: {
        default: () => setContextMenuId(-1),
        [STORY_CONTEXT_MENU_ACTIONS.COPY_STORY_LINK]: handleCopyStoryLink,
        [STORY_CONTEXT_MENU_ACTIONS.CREATE_TEMPLATE]: handleCreateTemplateFromStory,
        [STORY_CONTEXT_MENU_ACTIONS.DELETE]: handleDeleteStory,
        [STORY_CONTEXT_MENU_ACTIONS.DUPLICATE]: handleDuplicateStory,
        [STORY_CONTEXT_MENU_ACTIONS.OPEN_STORY_LINK]: handleOpenStoryInEditor,
        [STORY_CONTEXT_MENU_ACTIONS.RENAME]: handleRenameStory,
},```

Do y'all have any better ideas? Maybe we should keep the old way that feels like a reducer pattern.

Reply:
I do like the separation of concerns that you've implemented, but i can't help but wonder if the reducer pattern was better fit for the need of this menu specifically IN the story grid/list since you're having to generate functions to get the right data back and the tests are a little fragile. 

I love that the update you've made is flexible enough where this is easy enough to manage, especially thinking about future use cases where there's only 1 menu so everything's accessible (editor right click menu).

I don't want you to get stuck in this ticket. Let's see what @embarks / @littlemilkstudio think.

Reply:
an action map or reducer kinda feel the same to me. I'm pretty fine either way. 

If anything the `generateStoryMenu` kinda seems like a needless abstraction that adds some indirection. I'd be fine with the bulk and repetition of just defining the menu data in a `useMemo` vs having a function that iterates through things and connects it all together. idk tho, just feels like a bunch of ways to do the same thing with none particularly better than another imo. lol I'm not much help, just saying I'm fine with it as is 😄

Reply:
I think leave this how it is here and later we can revisit when we're not on as strict of a deadline for redesign stuff. I think we're going to get lost in the weeds a bit and that either one works just fine.

Reply:
I agree with you @littlemilkstudio. I think this abstraction makes it more difficult to figure out where the items come from.

I think I'll keep the `generateStoryMenu` in that file for now and maybe we can come back to it later